src/routes/profiles.py

```
@@ -1,5 +1,94 @@
+from database import get_db
+from config.dependencies import get_s3_storage_client, get_jwt_auth_manager
+from security.http import get_token
+from storages import S3StorageInterface
+from exceptions import TokenExpiredError, InvalidTokenError
+from schemas.profiles import ProfileRequestSchema, ProfileResponseSchema
+from validation import validate_image
 
 router = APIRouter()
 
-# Write your code here
+
+@router.post("/users/{user_id}/profile/", response_model=ProfileResponseSchema,
+             status_code=status.HTTP_201_CREATED)
+async def create_profile(
+    user_id: int,
+    form: ProfileRequestSchema = Depends(ProfileRequestSchema.as_form),
+    avatar: UploadFile = File(...),
+    token: str = Depends(get_token),
+    db: AsyncSession = Depends(get_db),
+    s3_client: S3StorageInterface = Depends(get_s3_storage_client),
+    jwt_manager=Depends(get_jwt_auth_manager),
+):
+    try:
+        payload = jwt_manager.decode_access_token(token)
+    except TokenExpiredError:
+        raise HTTPException(status_code=401, detail="Token has expired.")
+    except InvalidTokenError:
+        raise HTTPException(status_code=401, detail="Invalid or expired token")
+
+    user_id_from_token = payload.get("user_id")
+    if not user_id_from_token:
+        raise HTTPException(status_code=401, detail="Invalid or expired token")
+
+    user_id_from_token = int(user_id_from_token)
+
+    result = await db.execute(
+        select(UserModel).where(UserModel.id == user_id_from_token)
+    )
+    current_user = result.scalar_one_or_none()
+
+    if not current_user or not current_user.is_active:
+        raise HTTPException(status_code=401, detail="Invalid or expired token")
+
+    is_admin = current_user.group_id == 3
+    if not is_admin and current_user.id != user_id:
+        raise HTTPException(
+            status_code=403,
+            detail="You don't have permission to edit this profile."
+        )
+
```

There's a critical check missing here. According to the requirements, you must verify if the user specified by the `user_id` path parameter exists and is active. If the user is not found or is inactive, you should raise a 401 Unauthorized error with the detail: `"User not found or not active."`. This should be done before checking if a profile already exists.